### ISSUE TITLE
[infra] CherryPy docs project disappeared from RTD

### ISSUE THREAD

**Issue by webknjaz** [cherrypy/cherrypy#2013]
TL;DR https://docs.cherrypy.dev is currently defunct

I got a notification with UptimeRobot earlier today (I use it for monitoring some of the websites I manage). At first, I thought it's a small glitch or a problem between Read The Docs and CloudFlare, but I didn't do any changes to DNS on CF and I checked that the CNAME is correct.

I tried going to the RTD dashboard to check what's there and was surprised that it wasn't on the list of my projects. I figured that maybe @jaraco accidentally dropped my access, but https://readthedocs.org/projects/cherrypy/ (or https://readthedocs.org/projects/CherryPy/ for that matter) gives 404.

The webhook still exists in this repo settings and it indicates that the project id was `cherrypy`. But the `Recent deliveries` tab in that webhook doesn't list anything.

Not sure what's happening, but it points to either a platform problem on the RTD side or maybe Jason accidentally hit `Delete project`.

**Comment by jaraco** [comment]
I did not delete the project. My most recent email from rtd regarding cherrypy was in August, warning of a deprecation of "`python.sytem_packages` or `python.use_system_site_packages`".

I think the next step is to reach out to RTD admins and find out who might have deleted it.

**Comment by jaraco** [comment]
I've submitted a support request. I suggest we don't recreate the project until they've had some time to investigate. Presumably we can just recreate the project after we learn the cause.

**Comment by webknjaz** [comment]
Yes, did the same. Though, it doesn't seem like they send out ack-emails after filling out that form. FWIW the monitoring allows us to say that a day ago it was still functional.

**Comment by webknjaz** [comment]
Do we have a record who else had access to the RTD project? I think there were a few people but I haven't checked this specific RTD project in years. @Lawouach? @cyraxjoe? @tabo? @josephtate? @nicklasb? @aminusfu?
Could this be a consequence of one of the maintainer accounts being removed from RTD?

**Comment by Lawouach** [comment]
Hi all,

It's my fault. RTD pestered me because all my projects were going to be obsolete with some changes. So I deleted them as I don't use them, assuming the CP doc lived elsewhere. I'm an idiot for not asking here beforehand.

I'm happy to recreate the pj but since it was under my umbrella, would it make sense to do it under a CP account instead?

I really do apologise :(

Sylvain

**Comment by webknjaz** [comment]
@Lawouach thanks for the update! RTD doesn't have a concept of organizations and projects have "maintainers" with no privilege differences. Anybody can create a project and add others.
Though, I'm not sure if RTD would let us reuse the name because of their immutability stance. I'll try creating it but we may need to ask the support for help.

**Comment by webknjaz** [comment]
I managed to create a new project under the same identifier and invited @jaraco. I don't think anybody else wants/needs access so feel free to request if I'm missing something.

**Comment by webknjaz** [comment]
So far, build attempts get stuck. Maybe, it's because it didn't receive any `push` events from GH, not sure. I hacked it via DevTools to have `main` as the default branch but manually triggered builds still get stuck: https://readthedocs.org/projects/cherrypy/builds/23031179/.

**Comment by webknjaz** [comment]
Alright.. After poking it a bit more and retriggering a random PR, it caught up with the repo refs and started performing builds, which... fail.

I guess the integration is now functional, so I'll be able to get it to work.

**Comment by webknjaz** [comment]
The RTD docs site is back online. Closing.

**Comment by jaraco** [comment]
Thank you!

**Comment by nicklasb** [comment]
Kewl! I wondered what the heck was going on. :-)

**Comment by Lawouach** [comment]
Yeap thank you so much.

I really had no idea the docs was still under my perimeter. Specially as I
don't use RTD anymore. Mind you the build would have soon failed with their
new changes. Still, apologise once again folks.

Le ven. 5 janv. 2024, 20:19, Nicklas Börjesson ***@***.***> a
écrit :

**Comment by webknjaz** [comment]
> Mind you the build would have soon failed with their new changes.

That was not a problem for us. There was one runtime incompatibility unrelated to RTD (due to a recently accepted PR) but I fixed that too.

**Comment by Lawouach** [comment]
I see. Thanks for the info.
